Cargo.toml

```diff
@@ -20,6 +20,7 @@ exclude = ["datafusion-cli", "dev/depcheck"]
 members = [
     "datafusion/common",
     "datafusion/common-runtime",
+    "datafusion/catalog",
```

Comment:
nit: should we rename it to catalog-common, so we can reserve `catalog` for the actual catalog implementation

Reply:
pushed a change for this (https://github.com/apache/datafusion/pull/11516#issuecomment-2243214905)